draft-ietf-moq-transport.md

```diff
@@ -1806,6 +1797,77 @@ STREAM_HEADER_PEEP {
+
+The format of FETCH is as follows:
+
+~~~
+FETCH Message {
+  Track Namespace (b),
+  Track Name (b),
+  Priority (8),
+  Group Order (8),
+  StartGroup (i),
+  StartObject (i),
+  EndGroup (i),
+  EndObject (i),
+  Number of Parameters (i),
+  Fetch Parameters (..) ...
+}
+~~~
+{: #moq-transport-fetch-format title="MOQT FETCH Message"}
+
+* Track Namespace: Identifies the namespace of the track as defined in
+({{track-name}}).
+
+* Track Name: Identifies the track name as defined in ({{track-name}}).
+
+* Priority: Specifies the priority of a fetch request relative to
+other subscriptions or fetches in the same session. Lower numbers get higher priority. See {{priorities}}. It is typical of certain implementations to treat fetches to have lower priority than subscriptions to the live data.
+
+* Group Order: Allows the subscriber to request Objects be delivered in
+Ascending (0x1) or Descending (0x2) order by group. See {{priorities}}.
+A value of 0x0 indicates the original publisher's Group Order SHOULD be
+used. Values larger than 0x2 are a protocol error.
+
+* StartGroup: The start Group ID of the requested range.
+
+* StartObject: The start Object ID of the request range.
+
+* EndGroup: The end Group ID of the requested range.
+
+* EndObject: The end Object ID, plus 1 of the request range. A value of 0 means the entire group is requested.
+
+* Fetch Parameters: The parameters are defined in {{fetch-params}}.
+
+Fetch specifies a closed subscription starting at StartObject in StartGroup and ending at EndObject in EndGroup. The start and end of the range are inclusive. EndGroup and EndObject MUST specify the same or a later object than StartGroup and StartObject. If StartGroup/EndGroup is greater than the latest group at the publisher, the publisher MUST return FETCH_ERROR with INVALID_RANGE as the error code and close the stream.
+
+Otherwise the publisher responds with objects from the requested range over the same stream as the FETCH request. The behavior at the publisher is controlled by the `Delivery Expectation` fetch parameter. see {{fetch-params}}.
+
+The subscriber can also provide an hint in the fetch request, via the `Delivery Rate` parameter ({{fetch-params}}), to request the rate at which the data needs to be delivered by the publisher.
+
+### Fetch Parameters {#fetch-params}
+
+#### DELIVERY RATE Parameter {#fetch-delivery-rate}
+
+DELIVERY_RATE (key 0x01): An integer expressing the bitrate in number of bits per second specified in the FETCH message. If
+present, the publisher MUST attempt to delivery the objects at the rate requested or return FETCH_ERROR message with error code of "RATE_ERROR". If omitted, delivery rate informed by the underlying transport is choosen by the publisher.
+
+#### DELIVERY EXPECTATION Parameter {#fetch-delivery-expectation}
+
+FETCH_DELIVERY_EXPECTATION (key 0x02): An integeral enumeration expressing subscribers expectation of the publisher to fullfil a fetch reqeust.
```

Comment:
Fetch_Upstream (key): Boolean flag, true/1 requests to fetch from upstream any requested objects not in current cache, false/0 will only deliver requested objects from current cache.